.github/workflows/upgrade_test_constraints.yml

```diff
@@ -189,24 +176,16 @@ jobs:
           constraints="${prefix}/resources/constraints"
 
 
-          # allow to put in constraints things like setuptools (look at the end of one of the generated files). It will be the default behavior in the future.
-          flags+=" --allow-unsafe"
-
-          # pip constrains format does not allow to specify extras (like dask[array]) so we need to use this option
-          flags+=" --strip-extras"
-
-          # future default resolver. It is faster. Lower probability of long CI run.
-          flags+=" --resolver=backtracking"
-
           for pyv in 3.9 3.10 3.11 3.12; do
-            python${pyv}  -m pip install -U pip pip-tools
-            python${pyv}  -m piptools compile --upgrade -o $constraints/constraints_py${pyv}.txt  $pyproject_toml $constraints/version_denylist.txt ${flags}
-            python${pyv}  -m piptools compile --upgrade -o $constraints/constraints_py${pyv}_pydantic_1.txt  $pyproject_toml $constraints/version_denylist.txt $constraints/pydantic_le_2.txt ${flags}
+            uv pip compile --python-version ${pyv} --upgrade --output-file $constraints/constraints_py${pyv}.txt  $pyproject_toml $constraints/version_denylist.txt ${flags}
+            uv pip compile --python-version ${pyv} --upgrade --output-file $constraints/constraints_py${pyv}_pydantic_1.txt  $pyproject_toml $constraints/version_denylist.txt $constraints/pydantic_le_2.txt ${flags}
+            uv pip compile --python-platform aarch64-apple-darwin  --python-version ${pyv} --upgrade --output-file $constraints/constraints_py${pyv}_macos_arm.txt  $pyproject_toml $constraints/version_denylist.txt ${flags}
```

Comment:
So we no longer have a whole matrix, just a special case for macos?

Reply:
Yes. I started testing of post compile merge of files into one, but it will prevent from implementing the feature of updating constraints only for a given package. So I ended with an additional macOS arm file. I hope that we could drop it shortly.